packages/website/lib/api.ts

```diff
@@ -95,26 +116,131 @@ export async function getAllArticles(preview = false) {
 
   const articleEntries = extractArticleEntries(entries);
 
-  const sidebarLinks = articleEntries
-    .sort((a, b) => (a.title < b.title ? -1 : 1))
-    .reduce((acc, path) => {
-      const category = path.kbAppCategory?.slug ?? 'unassigned';
+  return articleEntries;
+}
+
+export async function getSidebarLinksBySectionSlug(
+  sectionSlug: string,
+  preview = false,
+) {
+  const entries = await fetchGraphQL(
+    `query {
+      sectionCollection(
+        where: { slug_exists: true, slug: "${sectionSlug}" }
+        limit: 1
+      ) {
+        items {
+          slug
+          title
+          sys {
+            id
+          }
+          linksCollection {
+            items {
+              ${LINKS_COLLECTION_GRAPHQL_FIELDS}
+            }
+          }
+          categoriesCollection {
+            items {
+              sys {
+                id
+              }
+              slug
+              name
+              type: __typename
+              linksCollection {
+                items {
+                  ${LINKS_COLLECTION_GRAPHQL_FIELDS}
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    `,
+    preview,
+  );
+
+  const data = entries?.data?.sectionCollection?.items[0];
 
-      const item = {
-        title: path.title,
-        slug: `/${path.kbAppCategory.slug}/${path.slug}`,
+  if (data) {
+    let sidebarLinks: SidebarSection[] = [];
+    const links = data.linksCollection?.items.map((link) => {
+      return {
+        ...link,
+        slug: link.slug.startsWith('http')
+          ? link.slug
+          : `/${sectionSlug}/${link.slug}`,
       };
+    });
 
-      if (acc[category]) {
-        return { ...acc, [category]: [...acc[category], item] };
-      }
+    if (links.length) {
+      sidebarLinks = [
+        {
+          links,
+        },
+      ];
+    }
 
-      return { ...acc, [category]: [item] };
-    }, {});
+    const categories = data.categoriesCollection?.items.reduce(
+      (categories, item) => {
+        const category = {
+          links: item.links,
+          slug: `/${sectionSlug}`,
+          title: item.name,
+        };
 
-  // create a JSON file with sidebar links for pages that come from Contentful
-  const data = JSON.stringify(sidebarLinks, null, 2);
-  fs.writeFileSync('utils/contentfulSidebarLinks.json', data);
+        // Do we actually use this?
```

Comment:
what's that? 🤔

Reply:
We're fetching the `previewDescription` field on `main` branch as well but I'm not sure we actually use it.

Reply:
let's try to remove it then?